pkg/utils/common.go

```diff
@@ -130,3 +135,101 @@ func MigrateToNewDirStructure(orig string) string {
 		return filepath.Join(git.WegoRoot, git.WegoAppDir, f[len(f)-2], f[len(f)-1])
 	}
 }
+
+type ConfigStatus int
+
+const (
+	Missing ConfigStatus = iota
+	Partial
+	Embedded
+	Valid
+)
+
+func (cs ConfigStatus) String() string {
+	switch cs {
+	case Missing:
+		return "Missing"
+	case Partial:
+		return "Partial"
+	case Embedded:
+		return "Embedded"
+	case Valid:
+		return "Valid"
+	default:
+		return "UnknownStatus"
+	}
+}
+
+type WalkResult struct {
+	Status ConfigStatus
+	Path   string
+}
+
+func (wr WalkResult) Error() string {
+	return fmt.Sprintf("found %s: with status: %s", wr.Path, wr.Status)
+}
+
+func FindCoreConfig(dir string) WalkResult {
```

Comment:
This function feels fragile. Could we just ask the user for the location instead?

Reply:
James and I discussed this a bit. The main reason for doing this is that I felt like the "single file containing both manifests and nothing else" case should be by far the most common. This provides a very simple upgrade experience for most users (who won't have to concern themselves with the core config at all. In particular, if someone upgrades six months after installing core (potentially not the same person who installed core) no knowledge would be required in the typical scenario.

As for  the fragility, except for the case in which a user put the core manifests in a single file _along with other manifests_ and _still_ wants to the delete the whole file, this function should be correct.

Reply:
That makes sense. I guess this is something we'll just have to make sure is clear in the documentation (both public & developer).